src/control/server/storage/bdev/provider.go

```diff
@@ -152,7 +154,18 @@ func (p *Provider) IsVMDDisabled() bool {
 func (p *Provider) Scan(req ScanRequest) (*ScanResponse, error) {
 	if p.shouldForward(req) {
 		req.DisableVMD = p.IsVMDDisabled()
-		return p.fwd.Scan(req)
+
+		if p.scanCache == nil || req.Rescan {
```

Comment:
not in usual operation but I guess there is nothing to prevent simultaneous requests from multiple dmg instances, I can add a mutex

Reply:
I think that would be safest. A simple Mutex rather than a RWMutex would probably be best, because you can't convert a read lock into a write lock, and dropping the read lock to get a write lock would be racy.